components/apimgt/org.wso2.carbon.apimgt.tracing/src/main/java/org/wso2/carbon/apimgt/tracing/TracingReporter.java

```diff
@@ -0,0 +1,124 @@
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.tracing;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import io.opentracing.contrib.reporter.LogLevel;
+import io.opentracing.contrib.reporter.Reporter;
+import io.opentracing.contrib.reporter.SpanData;
+import org.apache.commons.logging.Log;
+
+import java.io.StringWriter;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Iterator;
+import java.util.Map;
+
+public class TracingReporter implements Reporter {
+
+    private final Log log;
+    private final JsonFactory jsonFactory = new JsonFactory();
+
+    public TracingReporter(Log log) {
+        this.log = log;
+    }
+
+    public void start(Instant timeStamp, SpanData span) {
+    }
+
+    public void finish(Instant timeStamp, SpanData span) {
+        if (this.log.isTraceEnabled()) {
+            this.log.trace(this.toStructuredMessage(timeStamp, span));
+        }
+    }
+
+    public void log(Instant timeStamp, SpanData span, Map<String, ?> fields) {
+        LogLevel level = LogLevel.INFO;
+        try {
+            LogLevel level0 = (LogLevel) fields.get(LogLevel.FIELD_NAME);
+            if (level0 != null) {
+                level = level0;
+                fields.remove(LogLevel.FIELD_NAME);
+            }
+        } catch (Exception exc) {
+            log.warn("fail to read value of field {}");
+        }
+        switch (level) {
+            case TRACE:
+                if (log.isTraceEnabled()) {
+                    log.trace(toStructuredMessage(timeStamp, span));
+                }
+                break;
+            case DEBUG:
+                if (log.isDebugEnabled()) {
+                    log.debug(toStructuredMessage(timeStamp, span));
+                }
+                break;
+            case WARN:
+                if (log.isWarnEnabled()) {
+                    log.warn(toStructuredMessage(timeStamp, span));
+                }
+                break;
+            case ERROR:
+                if (log.isErrorEnabled()) {
+                    log.error(toStructuredMessage(timeStamp, span));
+                }
+                break;
+            default:
+                if (log.isInfoEnabled()) {
+                    log.info(toStructuredMessage(timeStamp, span));
+                }
+        }
+    }
+    //method for stucturing the tracer log format
```

Comment:
Shall we add method level JAVA doc comments for each method? So the code will be easy for anyone to understand.

Reply:
Fixed